kong/cmd/utils/prefix_handler.lua

```diff
@@ -211,7 +213,8 @@ local function prepare_prefix(kong_config, nginx_custom_template_path)
     kong_config.ssl_cert = kong_config.ssl_cert_default
     kong_config.ssl_cert_key = kong_config.ssl_cert_key_default
   end
-  if kong_config.admin_ssl and not kong_config.admin_ssl_cert and not kong_config.admin_ssl_cert_key then
+  if has_ssl and not kong_config.admin_ssl_cert and
```

Comment:
And here, we are using `has_ssl` which is computed over the proxy_listen values, not the Admin listen ones. We should probably use `kong_config.admin_ssl_enabled` anyway.

Reply:
There really should be tests covering this...